internal/packages/packagebytes/transform.go

```diff
@@ -0,0 +1,80 @@
+package packagebytes
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"text/template"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"package-operator.run/package-operator/internal/packages"
+)
+
+type Transformer interface {
+	Transform(ctx context.Context, fileMap FileMap) error
```

Comment:
This enforces that the FileMap will be modified in place. Unsure if that is good or a Signature like
`Transform(ctx context.Context, fileMap FileMap) (FileMap, error)` would be nicer to use for some applications

Reply:
This would require copying the FileMap over and over again. As this map can be a few MB in size, I would rather not clone it.